src/main/java/me/zeroeightsix/kami/feature/module/player/Blink.java

```diff
@@ -1,47 +1,79 @@
 package me.zeroeightsix.kami.feature.module.player;
 
+import io.github.fablabsmc.fablabs.api.fiber.v1.annotation.Setting;
 import me.zero.alpine.listener.EventHandler;
 import me.zero.alpine.listener.Listener;
 import me.zeroeightsix.kami.event.PacketEvent;
 import me.zeroeightsix.kami.feature.module.Module;
+import me.zeroeightsix.kami.util.Texts;
 import net.minecraft.client.network.OtherClientPlayerEntity;
 import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.network.Packet;
 import net.minecraft.network.packet.c2s.play.PlayerMoveC2SPacket;
+import net.minecraft.util.Formatting;
 
 import java.util.LinkedList;
 import java.util.Queue;
 
+
 @Module.Info(name = "Blink", category = Module.Category.PLAYER)
 public class Blink extends Module {
+    @Setting(name = "PacketDiscard")
+    private boolean packetDiscard = true;
+    @Setting(name = "WithholdAllPackets", comment = "Withhold all packets, not just PlayerMoveC2SPacket")
+    private boolean withholdAllPackets = true;
+    @Setting(name = "MaxPacketSize")
+    private @Setting.Constrain.Range(min = 50, max = 500, step = Double.MIN_VALUE) int maxPacketAmount = 100;
+
+    private OtherClientPlayerEntity clonedPlayer; // Fake Player when player blinked
+    Queue<Packet> packets = new LinkedList<>(); // Create list to hold our packets
 
-    Queue<PlayerMoveC2SPacket> packets = new LinkedList<>();
     @EventHandler
     public Listener<PacketEvent.Send> listener = new Listener<>(event -> {
-        if (event.getPacket() instanceof PlayerMoveC2SPacket) {
+        // We don't want to withhold login packets if a player logs out with blink enabled
+        if (event.getPacket() instanceof PlayerMoveC2SPacket || (withholdAllPackets && mc.world != null)) {
             event.cancel();
-            packets.add((PlayerMoveC2SPacket) event.getPacket());
+            packets.add(event.getPacket());
+
+            // Create a sense of urgency to unblick when approaching packet threshold
+            Formatting packetColor = Formatting.WHITE;
+            if (packets.size() > maxPacketAmount*(0.6f)) packetColor = Formatting.YELLOW;
+            if (packets.size() > maxPacketAmount) packetColor = Formatting.RED;
+
+            // Shows up above hotbar
```

Comment:
you should draw text on the screen instead of relying on some sort of formatting

Reply:
What do you mean

Reply:
This displays it in a nice textbox above the hotbar

Reply:
`mc.player.sendMessage` sends a new message each time, and this is dependent on the GUI scale and screen size...

Reply:
I know. Thats the point.

Reply:
What?

Reply:
The whole point is that it updates as every packet gets withheld.

Reply:
And that would spam the chat, which is why you should display the text on screen, instead of relying on something that won't display correctly the second you change the window size or gui scale

Reply:
It does **not** spam chat, by setting `bl` to `true`, it displays it in a snackbar and doesn't spam anything.